public/scripts/request.js

```diff
@@ -0,0 +1,41 @@
+export const newGameRequest = (id, state, callback) => {
+  let payload = jsonPostPayload;
+  payload.body = state;
+  payload.body.gameType = id;
+  payload.body = JSON.stringify(payload.body);
+  makeRequest(endpoints.newGame, payload, callback);
+}
+
+export const makeMoveRequest = (idx, state, callback) => {  
+  let payload = jsonPostPayload;
```

Comment:
This can be refactored using object destructuring:
```javascript
const payload = { ...jsonPostPayload, body: JSON.stringify({ ...state, selectedIdx: idx }) };
```